src/firebolt/async_db/cursor.py

```diff
@@ -366,6 +387,35 @@ async def _validate_set_parameter(self, parameter: SetParameter) -> None:
         # set parameter passed validation
         self._set_parameters[parameter.name] = parameter.value
 
+    def _validate_ss_async_settings(
+        self,
+        parameters: Sequence[Sequence[ParameterType]],
+        queries: List[Union[SetParameter, str]],
+        skip_parsing: bool = False,
+        async_execution: Optional[bool] = False,
+    ) -> None:
+        if async_execution and self._set_parameters.get("use_standard_sql", "1") == "0":
```

Comment:
Do we need to do a validation here ? (isn't it validated by backend?)

Reply:
I don't think so. I was having no issues setting these to incompatible values. In fact, it was making my debugging incredibly annoying.

Do you deal with this in the JDBC?

Reply:
I don't validate specific params with JDBC (once a param is set I call SELECT 1;). If it works, fine, if not then I return the error from backend .

Reply:
It doesn't actually return an error. Instead, the query id never gets written to the query_history table so we just get query not ready messages. I don't see a way around this other than checking in the SDK. I'm going to resolve this.